DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/igmp/test_igmp_snooping.py

```diff
@@ -0,0 +1,612 @@
+    macs = ["a6:00:00:00:00:01", "a6:00:00:00:00:02", "a6:00:00:00:00:03", "a6:00:00:00:00:04"]
+
+    # 1.Create a bridge and enable IGMP Snooping, enable querrier and change query interval
+    #   Config Fastleave on 1st rx_port
+    await common_bridge_and_igmp_setup(dev_name, bridge, 2, dut_ports, querier_interval=querrier_interval)
+
+    out = await BridgeLink.set(
+        input_data=[{dev_name: [
+            {"device": dut_ports[1], "fastleave": "on"}]}])
+    err_msg = f"Verify that port entities set to fastleave ON.\n{out}"
+    assert out[0][dev_name]["rc"] == 0, err_msg
+
+    # 2.Init interfaces and create 2 multicast Streams
+    # 3.Create 3 membership report streams, 1 with invalid checksum
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "102.0.0.3", "gw": "102.0.0.1", "plen": 24},
+         {"ixp": tg_ports[1], "ip": "100.0.0.3", "gw": "100.0.0.1", "plen": 24},
+         {"ixp": tg_ports[2], "ip": "101.0.0.3", "gw": "101.0.0.1", "plen": 24},
+         {"ixp": tg_ports[3], "ip": "99.0.0.3", "gw": "99.0.0.1", "plen": 24}])
+
+    tx_port = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"],
+                dev_groups[tg_ports[3]][0]["name"]]
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+    streams = {f"mcast_{idx + 1}": {
+            "type": "raw",
+            "protocol": "ip",
+            "ip_source": tx_port,
+            "ip_destination": rx_ports[1],
+            "srcMac": macs[0],
+            "dstMac": mcast_ip_to_mac(mcast_group),
+            "srcIp": "0.0.0.0",
+            "dstIp": mcast_group,
+            "frame_rate_type": "line_rate",
+            "rate": 40,
+        } for idx, mcast_group in enumerate([mcast_group1, mcast_group2])}
+
+    streams.update({f"member_report{idx + 1}": {
+            "type": "raw",
+            "protocol": "ip",
+            "ip_source": dev_groups[tg_ports[idx + 1]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[0]][0]["name"],
+            "srcMac": combined[1],
+            "dstMac": mcast_ip_to_mac(combined[0]),
+            "srcIp": dev_groups[tg_ports[idx + 1]][0]["ip"],
+            "dstIp": combined[0],
+            "ipproto": "igmpv2",
+            "totalLength": "28",
+            "igmpType": "22",
+            "igmpGroupAddr": combined[0],
+            "rate": 1,
+            "transmissionControlType": "fixedPktCount",
+            "frameCount": 1,
+        } for idx, combined in enumerate(zip(mcast_group_addr, macs[1:]))
+    })
+    streams["member_report3"]["igmpChecksum"] = "0"
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 4.Send Traffic from router port and from clients
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(querrier_interval + 5)

```

Comment:
Does the traffic stop if we remove cleanup_tgen fixture from the test and the test fails between traffic start and stop?

Reply:
Good point. If only a single test is run then I don't think that the traffic will stop. But if multiple tests are run then the next test will reset the session. Do we care that the traffic will still be running?